Makefile.uk

```diff
@@ -88,15 +88,32 @@ M4FLAGS      += -DUK_VERSION=$(UK_VERSION).$(UK_SUBVERSION)
 # If GCC supports "-no-pie" flag, we will add this flag to link flags to
 # override "pie" option, because some distributions will set
 # "--enable-default-pie" by default.
+ifeq ($(CONFIG_OPTIMIZE_PIE),y)
+COMPFLAGS-$(call gcc_version_ge,6,1)	+= -fPIE
+LDFLAGS-$(call gcc_version_ge,6,1)	+= -static-pie
```

Comment:
I need this to be `-static-pie` and not just `-pie`, because otherwise you are risking having GOT-related relocations generated which I do not and will not handle just to support an older version of the compiler because it does not make any sense to have this kind of relocations present in a kernel.

Reply:
My point was that you have to check for GCC >= 8 for this to work. If GCC < 8, you have to issue an error when `CONFIG_OPTIMIZE_PIE` is enabled.